src/blob/index/mod.rs

```diff
@@ -33,5 +33,5 @@ pub(crate) trait IndexTrait<K>: Send + Sync {
     fn count(&self) -> usize;
     async fn dump(&mut self, blob_size: u64) -> Result<usize>;
     async fn load(&mut self, blob_size: u64) -> Result<()>;
-    fn mark_all_as_deleted(&mut self, key: &K) -> Result<Option<u64>>;
+    fn mark_all_as_deleted(&mut self, key: &K, default_header: RecordHeader) -> Result<u64>;
```

Comment:
`default_header` is not a good name for parameter. It can be just `header` (`default` is misleading)

Reply:
Question: here you pass both `key` and `header` as parameters, but in `push` function above only `header` is passed. Is it possible to add `key` argument to `push` function?

Reply:
No, `push` is used during regeneration, and there are no keys at this moment.